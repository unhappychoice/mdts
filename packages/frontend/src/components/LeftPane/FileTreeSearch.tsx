import React from 'react';
import { Box, IconButton, InputAdornment, InputBase } from '@mui/material';
import ClearIcon from '@mui/icons-material/Clear';

interface FileTreeSearchProps {
  searchQuery: string;
  onSearchChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
  onClearSearch: () => void;
}

const FileTreeSearch: React.FC<FileTreeSearchProps> = ({ searchQuery, onSearchChange, onClearSearch }) => {
  return (
    <Box mb={2} px={2}>
      <InputBase
        placeholder="Search files..."
        inputProps={{ 'aria-label': 'search files' }}
        sx={{
          flex: 1,
          width: '100%',
          px: 2,
          py: 0,
          border: '1px solid',
          borderColor: 'divider',
          borderRadius: '4px',
          fontSize: '0.875rem',
        }}
        value={searchQuery}
        onChange={onSearchChange}
        endAdornment={
          searchQuery && (
            <InputAdornment position="end">
              <IconButton
                onClick={onClearSearch}
                edge="end"
                size="small"
                aria-label="clear search"
              >
                <ClearIcon sx={{ fontSize: '0.875rem', color: 'divider' }}/>
              </IconButton>
            </InputAdornment>
          )
        }
      />
    </Box>
  );
};

export default FileTreeSearch;
