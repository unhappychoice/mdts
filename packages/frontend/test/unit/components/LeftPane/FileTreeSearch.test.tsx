import { fireEvent, render, screen } from '@testing-library/react';
import React from 'react';
import FileTreeSearch from '../../../../src/components/LeftPane/FileTreeSearch';

describe('FileTreeSearch', () => {
  test('renders correctly with empty search query', () => {
    const { asFragment } = render(
      <FileTreeSearch searchQuery="" onSearchChange={jest.fn()} onClearSearch={jest.fn()} />
    );
    expect(asFragment()).toMatchSnapshot();
    expect(screen.getByPlaceholderText('Search files...')).toBeInTheDocument();
    expect(screen.queryByLabelText('clear search')).not.toBeInTheDocument();
  });

  test('renders correctly with a search query', () => {
    const { asFragment } = render(
      <FileTreeSearch searchQuery="test" onSearchChange={jest.fn()} onClearSearch={jest.fn()} />
    );
    expect(asFragment()).toMatchSnapshot();
    expect(screen.getByDisplayValue('test')).toBeInTheDocument();
    expect(screen.getByLabelText('clear search')).toBeInTheDocument();
  });

  test('calls onSearchChange when input value changes', () => {
    const onSearchChangeMock = jest.fn();
    render(
      <FileTreeSearch searchQuery="" onSearchChange={onSearchChangeMock} onClearSearch={jest.fn()} />
    );
    fireEvent.change(screen.getByPlaceholderText('Search files...'), { target: { value: 'new query' } });
    expect(onSearchChangeMock).toHaveBeenCalledWith(expect.any(Object));
  });

  test('calls onClearSearch when clear button is clicked', () => {
    const onClearSearchMock = jest.fn();
    render(
      <FileTreeSearch searchQuery="test" onSearchChange={jest.fn()} onClearSearch={onClearSearchMock} />
    );
    fireEvent.click(screen.getByLabelText('clear search'));
    expect(onClearSearchMock).toHaveBeenCalled();
  });
});
