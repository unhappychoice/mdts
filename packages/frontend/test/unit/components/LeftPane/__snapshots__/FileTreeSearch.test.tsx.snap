// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`FileTreeSearch renders correctly with a search query 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-kuuj3l"
  >
    <div
      class="MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-adornedEnd css-pe16f2-MuiInputBase-root"
    >
      <input
        aria-label="search files"
        class="MuiInputBase-input MuiInputBase-inputAdornedEnd css-e77w8q-MuiInputBase-input"
        placeholder="Search files..."
        type="text"
        value="test"
      />
      <div
        class="MuiInputAdornment-root MuiInputAdornment-positionEnd css-elo8k2-MuiInputAdornment-root"
      >
        <button
          aria-label="clear search"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeSmall css-1eswsun-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-kqgmkm-MuiSvgIcon-root"
            data-testid="ClearIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`FileTreeSearch renders correctly with empty search query 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-kuuj3l"
  >
    <div
      class="MuiInputBase-root MuiInputBase-colorPrimary css-pe16f2-MuiInputBase-root"
    >
      <input
        aria-label="search files"
        class="MuiInputBase-input css-e77w8q-MuiInputBase-input"
        placeholder="Search files..."
        type="text"
        value=""
      />
    </div>
  </div>
</DocumentFragment>
`;
